### ISSUE TITLE
Noncommutative multiplication with string argument triggers "declare: argument must be a symbol"

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2494]
Imported from SourceForge on 2024-07-06 10:03:54
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-01 21:15:45
Original: https://sourceforge.net/p/maxima/bugs/3935

---

`a . "b"` and  `"a" . b`  both trigger the error message "declare: argument must be a symbol". 

Some debugging shows that's coming out of KINDP:
```
(%i2) "a".b;
1. Trace: (KINDP '"a" '$CONSTANT)
declare: argument must be a symbol; found a
```
I think KINDP should just return NIL in this case; I don't think KINDP or SIMPNCT should trigger an error for this case.

For the record, I see that `"a" + 1` and `2*"a"` don't trigger errors. I think that's okay and I propose that we keep it that way.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:03:55
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-04 19:23:18
Original: https://sourceforge.net/p/maxima/bugs/3935/#b092

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:03:59
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-04 19:23:19
Original: https://sourceforge.net/p/maxima/bugs/3935/#d5b8

---

Fixed by commit f6cf9e5, closing this report.
